config/config/pom.xml

```diff
@@ -42,6 +42,10 @@
             <groupId>jakarta.annotation</groupId>
             <artifactId>jakarta.annotation-api</artifactId>
         </dependency>
+        <dependency>
```

Comment:
This feels backwards.

The declarative API should be on layered on top of the programmatic API, right ?
As it stands `helidon-config` pulls `pico-config` even if pico isn't used.

Reply:
There is a chicken-egg problem with config. It can be solved two ways:
1. create a config API that is implemented by `helidon-config` and backward compatible
2. make `helidon-config` the config API and have a separate implementation module (not backward compatible)

The first option seems to have the least impact on Helidon